Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeConnectors.ps1

```diff
@@ -0,0 +1,219 @@
+                $exchangeFactoryConnectorReturnObject.CloudEnabled = $ConnectorObject.CloudServicesMailEnabled
+                if ($null -ne $ConnectorObject.TlsAuthLevel) {
+                    $exchangeFactoryConnectorReturnObject.TlsAuthLevel = $ConnectorObject.TlsAuthLevel
+                }
+            }
+
+            if ($null -ne $ConnectorObject.TlsCertificateName) {
+                Write-Verbose "TlsCertificateName is configured on this connector"
+                $exchangeFactoryConnectorReturnObject.TlsCertificateSet = $true
+                $exchangeFactoryConnectorReturnObject.TlsCertificateName = ($ConnectorObject.TlsCertificateName).ToString()
+            } else {
+                Write-Verbose "TlsCertificateName is not configured on this connector"
+                $exchangeFactoryConnectorReturnObject.TlsCertificateNameStatus = "TlsCertificateNameEmpty"
+            }
+
+            return $exchangeFactoryConnectorReturnObject
+        }
+
+        function NormalizeTlsCertificateName {
+            [CmdletBinding()]
+            [OutputType("System.Object")]
+            param(
+                [Parameter(Mandatory = $true)]
+                [string]
+                $TlsCertificateName
+            )
+
+            Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+            try {
+                Write-Verbose ("TlsCertificateName that was passed: '{0}'" -f $TlsCertificateName)
+                # RegEx to match the recommended value which is "<I>X.500Issuer<S>X.500Subject"
+                if ($TlsCertificateName -match "(<i>).*(<s>).*") {
+                    $expectedTlsCertificateNameDetected = $true
+                    $issuerIndex = $TlsCertificateName.IndexOf("<I>", [System.StringComparison]::OrdinalIgnoreCase)
+                    $subjectIndex = $TlsCertificateName.IndexOf("<S>", [System.StringComparison]::OrdinalIgnoreCase)
+
+                    Write-Verbose "TlsCertificateName that matches the expected syntax was passed"
+                } else {
+                    $issuerIndex = $TlsCertificateName.IndexOf("CN=", [System.StringComparison]::OrdinalIgnoreCase)
+                    $subjectIndex = $TlsCertificateName.LastIndexOf("CN=", [System.StringComparison]::OrdinalIgnoreCase)
+
+                    Write-Verbose "TlsCertificateName with bad syntax was passed"
+                }
+
+                if (($issuerIndex -ne -1) -and
+                    ($subjectIndex -ne -1)) {
+                    if ($expectedTlsCertificateNameDetected) {
+                        $issuer = $TlsCertificateName.Substring(($issuerIndex + 3), ($subjectIndex - 3))
+                        $subject = $TlsCertificateName.Substring($subjectIndex + 3)
+                    } else {
+                        $issuer  = $TlsCertificateName.Substring($issuerIndex, $subjectIndex)
+                        $subject = $TlsCertificateName.Substring($subjectIndex)
```

Comment:
If `$issuerIndex` or `$subjectIndex` is set to a `-1` value this will throw an error. 

![image](https://user-images.githubusercontent.com/22776718/161543538-1b220284-f257-4020-9ea5-c41e53dabb98.png)

Reply:
Yep, therefore I've placed it inside this if statement (or have I misunderstood something?):
<img width="457" alt="image" src="https://user-images.githubusercontent.com/40993616/161547163-ed5b63fd-20b7-4ac3-87b2-68edb6d413de.png">

Reply:
That if statement requires both `$issuerIndex` AND `$subjectIndex` to not equal `-1`. What about when only 1 of those variables are set to `-1`?

Reply:
We don't have enough information available when only one of both has a value other than -1. In this case `NormalizeTlsCertificateName` doesn't return anything. However, this case should not occur in normal circumstances as it's not possible to set any other value than `$null` or `(<i>).*(<s>).*` for `TlsCertificateName` via PowerShell.  It's more a failsafe and we normally should not end up here.

Reply:
Right, and what I am saying is that you will cause an error to occur here if one of the values are `-1`. Thus, the `else` statement wouldn't really execute the way you are wanting.

Reply:
Ah, I know what you mean. The idea of the `else `statement is to handle this particular case (which should also not occur - just a failsafe): 

<img width="636" alt="image" src="https://user-images.githubusercontent.com/40993616/161578419-0bd7cde4-2f86-4eed-b125-be33bab8e1d6.png">
`$tlscertificatename = "$($cert.Issuer)$($cert.Subject)"` (without <I> and <S>)

Reply:
Added a comment to the script as this is by design.